packages/cerebral-provider-forms/src/form.test.js

```diff
@@ -0,0 +1,288 @@
+/* eslint-env mocha */
+import {Controller} from 'cerebral'
+import FormsProvider from '.'
+import assert from 'assert'
+import {Form, Field} from './form'
+
+describe('form', () => {
+  it('should create fields with default state', () => {
+    const controller = Controller({
+      providers: [FormsProvider()],
+      state: {
+        form: {
+          name: {
+            value: 'Ben'
+          }
+        }
+      },
+      signals: {
+        test: [
+          ({forms}) => {
+            const form = forms.get('form')
+            assert.ok(form instanceof Form)
+            assert.ok(form.name instanceof Field)
+            assert.equal(form.name.value, 'Ben')
+            assert.equal(form.name.isValid, true)
+          }
+        ]
+      }
+    })
+    controller.getSignal('test')()
+  })
+  it('should validate initial form state', () => {
+    const controller = Controller({
+      providers: [FormsProvider()],
+      state: {
+        form: {
+          name: {
+            value: 'Ben',
+            defaultValue: 'Ben',
+            validationRules: ['minLength:4']
+          }
+        }
+      },
+      signals: {
+        test: [
+          ({forms}) => {
+            const form = forms.get('form')
+            assert.ok(form instanceof Form)
+            assert.ok(form.name instanceof Field)
+            assert.equal(form.name.value, 'Ben')
+            assert.equal(form.name.isValid, false)
+            assert.deepEqual(form.name.validationRules, ['minLength:4'])
+            assert.equal(form.name.requiredMessage, null)
```

Comment:
looks like some leftover. `requiredMessage` is not set anywhere and thus `undefined`